### PR TITLE
Ninja Refactor

### PR DESCRIPTION
**Pull request by lordpidey**
This will move ninjas, and their incredibly snowflakey code into a standardized antagonist datum, as well as move ninja verbs into equipment based abilities.

For admins, this will add ninjas to the traitor panel, instead of putting them in the view variables of a mob, putting them in line with other antagonists.

:cl: Lordpidey
tweak: Space ninjas now use action buttons instead of verbs for a more consistent user experience.
/:cl:

Fixes: #11289

### CODE REVIEW

**Comment by Cyberboss** [comment]
Doing gods work

**Comment by banthebantz** [comment]
you absolute madman

**Comment by Anonmare** [comment]
I pray one day some madman will unsnowflake Ninja code. I find it really amusing that I can't make someone a space ninja or spawn myself in as one manually, and by amusing I mean annoying

**Comment by lordpidey** [comment]
This SHOULD unsnowflake a great deal of it.  The equipment is still going to be in it's own corner of the codebase, I'm not going to touch that.

**Comment by Armhulen** [comment]
HNNNNG NINJAAAAAAA

**Comment by Armhulen** [comment]
i'm spent

**Comment by Bawhoppen** [comment]
it's slowly happening

**Comment by CPTANT** [comment]
The ninja is a relic from a time long past.

**Comment by Armhulen** [comment]
@CPTANT ninjas aren't a thing now and they aren't in 2557

**Comment by Dorsisdwarf** [comment]
One of these days there will be a nicer way to do this than lumping every odd antag into "was the traitor"

**Comment by lordpidey** [comment]
Okay.  I've finished the changes I wanted to do, and quick testing didn't reveal any bugs.  I still need to test it more thoroughly.  But It's ready for review now.

Keep in mind, I didn't do a total re-write, I only focused on the abilities and datum itself.

File: code/modules/ninja/suit/suit.dm [hunk 1]
```diff
@@ -109,40 +110,38 @@ Contents:
...
+	n_shoes.flags |= NODROP
+	n_shoes.slowdown--
+	n_gloves = H.gloves
+	n_gloves.flags |= NODROP
+	return TRUE
+
+/obj/item/clothing/suit/space/space_ninja/proc/lockIcons(mob/living/carbon/human/H)
+	icon_state = H.gender==FEMALE ? "s-ninjanf" : "s-ninjan"
+	H.gloves.icon_state = "s-ninjan"
+	H.gloves.item_state = "s-ninjan"
```
Comment (Shadowlight213) [inline code/modules/ninja/suit/suit.dm]: Dupe?

File: code/datums/action.dm [hunk 2]
```diff
@@ -358,6 +358,58 @@
 		active = FALSE
 	..()
 
+/datum/action/item_action/initialize_ninja_suit
+	name = "Toggle ninja suit"
+
+/datum/action/item_action/ninjajaunt
+	name = "Phase Jaunt (10E)"
+	desc = "Utilizes the internal VOID-shift device to rapidly transit in direction facing."
+	button_icon_state = "jaunt"
```
Comment (ChangelingRain) [inline code/datums/action.dm]: take the ninja jaunt effect from effects.dmi

File: code/datums/action.dm [hunk 3]
```diff
@@ -358,6 +358,58 @@
...
+
+/datum/action/item_action/ninjaboost
+	name = "Adrenaline Boost"
+	desc = "Inject a secret chemical that will counteract all movement-impairing effect."
+	button_icon_state = "repulse"
+
+/datum/action/item_action/ninjapulse
+	name = "EM Burst (25E)"
+	desc = "Disable any nearby technology with a electro-magnetic pulse."
+	button_icon_state = "Sentinel's Compromise"
```
Comment (ChangelingRain) [inline code/datums/action.dm]: use the emp action state holy shit

File: code/datums/action.dm [hunk 4]
```diff
@@ -358,6 +358,58 @@
...
+
+/datum/action/item_action/ninja_stealth
+	name = "Toggle Stealth"
+	desc = "Toggles stealth mode on and off."
+	button_icon_state = "beam_up"
+
+/datum/action/item_action/toggle_glove
+	name = "Toggle interaction"
+	desc = "Switch between normal interaction and drain mode."
+	button_icon_state = "lightning"
```
Comment (ChangelingRain) [inline code/datums/action.dm]: maybe use the ninja gloves icon?

File: code/datums/action.dm [hunk 5]
```diff
@@ -358,6 +358,58 @@
...
+/datum/action/item_action/ninja_sword_recall
+	name = "Recall Energy Katana (Variable Cost)"
+	desc = "Teleports the Energy Katana linked to this suit to its wearer, cost based on distance."
+	button_icon_state = "energy_katana"
+	icon_icon = 'icons/obj/weapons.dmi'
+
+/datum/action/item_action/ninja_stealth
+	name = "Toggle Stealth"
+	desc = "Toggles stealth mode on and off."
+	button_icon_state = "beam_up"
```
Comment (ChangelingRain) [inline code/datums/action.dm]: use one of the ninja uncloak effects(cloak/uncloak) from mob.dmi

File: code/datums/action.dm (hunk 2 above)
Comment (ChangelingRain) [inline code/datums/action.dm]: rather, use one of the frames from one of those icon states

File: code/datums/action.dm (hunk 5 above)
Comment (ChangelingRain) [inline code/datums/action.dm]: same here, use one of the frames of those states

**Comment by lordpidey** [comment]
Aright, I've made those changes, @ChangelingRain, I can test them when I return home.

File: code/datums/action.dm [hunk 6]
```diff
@@ -379,7 +380,8 @@
 /datum/action/item_action/ninjapulse
 	name = "EM Burst (25E)"
 	desc = "Disable any nearby technology with a electro-magnetic pulse."
-	button_icon_state = "Sentinel's Compromise"
+	button_icon_state = "emppulse"
```
Comment (ChangelingRain) [inline code/datums/action.dm]: there's an emp icon state in actions.dmi

File: code/datums/action.dm [hunk 7]
```diff
@@ -364,7 +364,8 @@
 /datum/action/item_action/ninjajaunt
 	name = "Phase Jaunt (10E)"
 	desc = "Utilizes the internal VOID-shift device to rapidly transit in direction facing."
-	button_icon_state = "jaunt"
+	button_icon_state = "phasein"
```
Comment (ChangelingRain) [inline code/datums/action.dm]: use ninja_jaunt in actions.dmi

File: code/datums/action.dm [hunk 8]
```diff
@@ -402,12 +404,13 @@
 /datum/action/item_action/ninja_stealth
 	name = "Toggle Stealth"
 	desc = "Toggles stealth mode on and off."
-	button_icon_state = "beam_up"
+	button_icon_state = "ninja_cloak"
```
Comment (ChangelingRain) [inline code/datums/action.dm]: as note, ninja_cloak and ninja_jaunt should both probably not be animated, as it's distracting.

File: code/modules/ninja/suit/suit_initialisation.dm [hunk 9]
```diff
@@ -1,102 +1,76 @@
...
-					remove_equip_verbs()
-					unlock_suit()
-					U.regenerate_icons()
-			sleep(delay)
-		s_busy = 0
-	return
\ No newline at end of file
+			return FALSE
+		s_busy = TRUE
+		sleep(delay)
```
Comment (Really-Good-Soda-Flavor) [inline code/modules/ninja/suit/suit_initialisation.dm]: AAAAAAAAAAAAAAHHHHHHHHHHHHH There has to be a better way to do this than all of these sleeps

Comment (kevinz000) [inline code/modules/ninja/suit/suit_initialisation.dm]: addtimer

Comment (lordpidey) [inline code/modules/ninja/suit/suit_initialisation.dm]: addtimer is equivalent to spawn, not sleep.

I mean, I suppose I could do this with addtimer, but wouldn't that induce more overhead than a sleep?

**Comment by Cyberboss** [comment]
@lordpidey, no, in fact you'd incurr LESS because byond is dumb

**Comment by lordpidey** [comment]
.... huh.   Okay.   I'll change that to addtimers.

**Comment by RemieRichards** [comment]
>ugly

I'll slap you with a fish mike.

**Pull request merged**